### PR TITLE
Make more functions use ctx instead of db.DefaultContext

### PR DESCRIPTION
**Pull request by wxiaoguang**
Continue the "ctx refactoring" work.

There are still a lot db.DefaultContext, incorrect context could cause database deadlock errors.

### CODE REVIEW

File: services/issue/assignee.go [hunk 1]
```diff
@@ -44,25 +43,25 @@ func DeleteNotPassedAssignee(issue *issues_model.Issue, doer *user_model.User, a
 }
 
 // ToggleAssignee changes a user between assigned and not assigned for this issue, and make issue comment for it.
-func ToggleAssignee(issue *issues_model.Issue, doer *user_model.User, assigneeID int64) (removed bool, comment *issues_model.Comment, err error) {
+func ToggleAssignee(ctx context.Context, issue *issues_model.Issue, doer *user_model.User, assigneeID int64) (removed bool, comment *issues_model.Comment, err error) {
 	removed, comment, err = issues_model.ToggleIssueAssignee(issue, doer, assigneeID)
```
Comment (lunny) [inline services/issue/assignee.go]: This function `ToggleIssueAssignee` should also have a `ctx` parameter.

File: services/issue/assignee.go [hunk 2]
```diff
@@ -44,25 +43,25 @@ func DeleteNotPassedAssignee(issue *issues_model.Issue, doer *user_model.User, a
...
 
-	assignee, err1 := user_model.GetUserByID(db.DefaultContext, assigneeID)
+	assignee, err1 := user_model.GetUserByID(ctx, assigneeID)
 	if err1 != nil {
 		err = err1
 		return
 	}
 
-	notification.NotifyIssueChangeAssignee(db.DefaultContext, doer, issue, assignee, removed, comment)
+	notification.NotifyIssueChangeAssignee(ctx, doer, issue, assignee, removed, comment)
```
Comment (lunny) [inline services/issue/assignee.go:58]: Notify could be failed, so it should not has the same ctx.

Comment (wxiaoguang) [reply services/issue/assignee.go:58]: Even if `Notify` fails, how does it affect the ctx?

And I can see other code also does so (although I do not think it is ideal, either)

File: services/issue/assignee.go (hunk 1 above)
Comment (wxiaoguang) [reply services/issue/assignee.go]: Done in d528337bc366afb53a36e8736605e098fc2ed6b3

File: services/issue/assignee.go (hunk 2 above)
Comment (delvh) [reply services/issue/assignee.go:58]: Hmm… What can we do about it?
A separate `notification` context?
(But not in this PR)

File: services/issue/comments.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func CreateRefComment(doer *user_model.User, repo *repo_model.Repository, issue
 		return nil
 	}
 
-	_, err = CreateComment(&issues_model.CreateCommentOptions{
+	_, err = CreateComment(db.DefaultContext, &issues_model.CreateCommentOptions{
```
Comment (delvh) [inline services/issue/comments.go:56]: Hmm… Or we simply give this function a `ctx` as well…

File: services/issue/issue.go [hunk 4]
```diff
@@ -20,57 +21,57 @@ import (
...
 	return nil
 }
 
 // ChangeTitle changes the title of this issue, as the given user.
-func ChangeTitle(issue *issues_model.Issue, doer *user_model.User, title string) (err error) {
+func ChangeTitle(ctx context.Context, issue *issues_model.Issue, doer *user_model.User, title string) (err error) {
 	oldTitle := issue.Title
 	issue.Title = title
 
 	if err = issues_model.ChangeIssueTitle(issue, doer, oldTitle); err != nil {
```
Comment (delvh) [inline services/issue/issue.go]: No `ctx`?

**Review by delvh — APPROVED** [review]

File: services/issue/comments.go (hunk 3 above)
Comment (wxiaoguang) [reply services/issue/comments.go:56]: I have tried but it causes too many changes.

So just make it like [the line above](https://github.com/go-gitea/gitea/pull/24068/files/d528337bc366afb53a36e8736605e098fc2ed6b3#diff-cacc2474f9155102033f0df86a83e1c5019830bed4118c352f8bd6ff8c441c8bL45) (also use db.DefaultContext)

Leave it to next round refactoring.

File: services/issue/issue.go (hunk 4 above)
Comment (wxiaoguang) [reply services/issue/issue.go]: Good catch, done in dda1754ce1dee2eede3797bf9c8edce2795c8fa1

**Review by lunny — APPROVED** [review]

**Comment by techknowlogick** [comment]
🤖 🎺

**Pull request merged**